android/src/main/kotlin/com/f2fk/geofence_foreground_service/GeofenceForegroundServicePlugin.kt

```diff
@@ -68,134 +67,86 @@ class GeofenceForegroundServicePlugin : FlutterPlugin, MethodCallHandler, Activi
+            contentTitle = call.argument<String>(Constants.contentTitle)
+            contentText = call.argument<String>(Constants.contentText)
+            serviceId = call.argument<Int>(Constants.serviceId)
+            isInDebugMode = call.argument<Boolean>(Constants.isInDebugMode) ?: false
 
-                addGeofence(zone, result)
+            val iconDataJson: Map<String, Any>? = call.argument<Map<String, Any>>(
+                Constants.iconData
+            )
+            if (iconDataJson != null) {
+                iconData = NotificationIconData.fromJson(iconDataJson)
             }
+            serviceIntent = serviceIntent(GeofenceServiceAction.SETUP.toString())
+            val channel = NotificationChannel(
+                channelId,
+                "Geofence foreground service",
+                NotificationManager.IMPORTANCE_HIGH
+            )
+            channel.description = "A channel for receiving geofencing notifications"
 
-            "addGeoFences" -> {
-                val zonesList: ZonesList = ZonesList.fromJson(call.arguments as Map<String, Any>)
-
-                addGeoFences(zonesList, result)
-            }
+            val notificationManager =
+                activity?.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
 
-            "removeGeofence" -> {
-                val zonesId: String = call.argument(Constants.zoneId)!!
+            notificationManager.createNotificationChannel(channel)
 
-                removeGeofence(listOf(zonesId), result)
-            }
+            ContextCompat.startForegroundService(context, serviceIntent)
+            result.success(true)
+        } catch (e: Exception) {
+            result.success(false)
+        }
+    }
 
-            else -> {
-                result.notImplemented()
-            }
+    private fun stopGeofencingService(result: Result) {
+        try {
+            context.stopService(serviceIntent)
+            result.success(true)
+        } catch (e: Exception) {
+            result.success(false)
         }
     }
 
+    private fun serviceIntent(geofenceAction: String) : Intent =
+        Intent(context, GeofenceForegroundService::class.java)
+            .putExtra(activity!!.extraNameGen(Constants.isInDebugMode), isInDebugMode)
+            .putExtra(activity!!.extraNameGen(Constants.geofenceAction), geofenceAction)
+            .putExtra(activity!!.extraNameGen(Constants.appIcon), getIconResId(iconData))
+            .putExtra(activity!!.extraNameGen(Constants.channelId), channelId)
+            .putExtra(activity!!.extraNameGen(Constants.contentTitle), contentTitle)
+            .putExtra(activity!!.extraNameGen(Constants.contentText), contentText)
+            .putExtra(activity!!.extraNameGen(Constants.serviceId), serviceId)
+
+    // Build the pending intent for the intent service
+    private fun servicePendingIntent(intent: Intent) : PendingIntent = PendingIntent.getForegroundService(
```

Comment:
Also here, in the old code I was checking for the android OS above android O or not (if true then use getForegroundService else getService)